blog/tests.py

```diff
@@ -118,3 +132,172 @@ def test_delete_blog(self):
         force_authenticate(request, user=user)
         response = view(request, slug=blog.slug)
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_update_blog(self):
+        """
+        Tests to update a Blog Post Successfully.
+        """
+        response = self.create_blog()
+        user = User.objects.get(username='test')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        blog = Blog.objects.get()
+
+        body = {
+            'title': 'Test101',
+            'description': 'This is a Test Blog. Amirite! :D'
+        }
+
+        factory = APIRequestFactory()
+        url = '/api/blog/'
+
+        request = factory.put(url, body)
+        view = BlogAPI.as_view({
+            'put': 'update'
+        })
+
+        force_authenticate(request, user=user)
+        response = view(request, slug=blog.slug)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(Blog.objects.count(), 1)
+        self.assertEqual(Blog.objects.get().title, 'Test101')
+
+    def test_search_blog_list(self):
+        """
+        Tests to search Blog Posts Successfully.
+        """
+        response = self.create_blog()
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        url = '/api/blog/?keyword=es'
+        factory = APIRequestFactory()
+
+        request = factory.get(url)
+        view = BlogAPI.as_view({
+            'get': 'list'
+        })
+        response = view(request)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['count'], 1)
+        self.assertEqual(response.data['results'][0]['title'], 'Test')
+
+    def test_get_blog_list(self):
+        """
+        Tests to get Blog Posts Successfully.
+        """
+
+        url = '/api/blog/'
+        factory = APIRequestFactory()
+
+        request = factory.get(url)
+        view = BlogAPI.as_view({
+            'get': 'list'
+        })
+        response = view(request)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data['count'], 0)
+
+    def test_blog_upvote(self):
+        """
+        Tests Upvote functionality.
+        """
+        response = self.create_blog()
+        user = User.objects.get(username='test')
+
+        blog = Blog.objects.get()
+
+        factory = APIRequestFactory()
+        url = '/api/blog/'
+        request = factory.get(url)
```

Comment:
we can create a helper method for getting blog api factory or maybe move it in `setup`

Reply:
I've moved it to the class level attribute hence it is shared in every test.